### PR TITLE
Add Snippets

### LINKED ISSUE #51: Adding Snippets
Another useful feature to have. I personally like ISE way (ctr + j), but VSCode is good too (just typing snippet name and it appears on the autocomplete list).

### PR DESCRIPTION
**Pull request by corbob**
Fixes #51 

Still need to actually create the snippets entry in `Package.json`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 *.zip
 PowerShellEditorServices/
+Snippets/
```
Comment (yatli) [inline .gitignore]: can we ship the snippets directly? :)

Comment (corbob) [reply .gitignore]: I like the idea of piggy-backing off of the work done for vscode-powershell, and when the package.json is updated to have the snippets in there, it should ship them with the npm...

That being said, I think we can contribute multiple snippet files, so perhaps we can include a snippets file of our own?

Comment (yatli) [reply .gitignore]: it makes sense to allow for multiple entries, or even custom entries in user-defined directories.
I'll poke around to see if there's a way to programmatically add snippets without populating package.json ahead of time.

Comment (TylerLeonhardt) [reply .gitignore]: @yatli user-defined snippets is probably already a thing provided by coc.

I agree with @corbob that grabbing the vscode-powershell snippets is a good start here. If we want to add more on top of that, we can for sure in one of the other mechanisms

Comment (mikeTWC1984) [reply .gitignore]: I think we can also create quick ps module that would be adding snippets to PowerShell.json, so users can either update it manually or add their custom snippets. Like New-VimSnippet/Remove-VimSnippet/Import-VimSnippet (similar to ISE)

**Comment by TylerLeonhardt** [comment]
@Corbob do you wanna move this out of draft now? Is it ready?

File: .gitignore (hunk 1 above)
Comment (TylerLeonhardt) [reply .gitignore]: I know vscode has a concept of user-defined snippets, I wouldn't be surprised if coc.nvim has it as well.

That'd be the best place a `New-VimSnippet` could target by default.

**Comment by corbob** [comment]
> 
> 
> @corbob do you wanna move this out of draft now? Is it ready?

I think it's mostly good. The only thing I think should be added is a note to the readme about needing coc-snippets. 

I'll update it later tonight or tomorrow and then move it out of draft.

**Comment by corbob** [comment]
I did just notice a note that this needs nvim 0.4 or the latest vim8: https://github.com/neoclide/coc.nvim/wiki/F.A.Q#how-to-make-preview-window-shown-aside-with-pum which I've mentioned in the readme.

**Comment by yatli** [comment]
LGTM!

I've had a peek into coc-snippets, this is how it scans for snippets:

https://github.com/neoclide/coc-snippets/blob/bcfc9ecbf5e31e89a321a2f6e6bf6c3042aef166/src/textmateProvider.ts#L54-L66

So, if we modify the configuration setting `snippets.textmateSnippetsRoot` on activation, we make our snippets directory dynamic, and need not to prescribe it in package.json

**Comment by yatli** [comment]
Let's merge the current version first. I'm poking around this idea and we can do it in another PR.

**Pull request merged**